packages/utils/src/fulfillment/provider.ts

```diff
@@ -0,0 +1,69 @@
+import { IFulfillmentProvider } from "@medusajs/types"
+
+export class AbstractFulfillmentProviderService
+  implements IFulfillmentProvider
+{
+  static identifier: string
+
+  static _isFulfillmentService = true
+
+  static isFulfillmentService(obj) {
+    return obj?.constructor?._isFulfillmentService
+  }
+
+  getIdentifier() {
+    return (this.constructor as any).identifier
+  }
+
+  async getFulfillmentOptions(): Promise<Record<string, unknown>[]> {
```

Comment:
I can see the file is called provider, by the methods `Options`, so we definitely need to settle for one.

Reply:
this is just the base provider class a user need to use to create a fulfillment provider, but the options are the fulfillment options, not the provider "options" in the sense of config but in the sense of fulfillment, for example
- ups
- laposte
- etc